### PR TITLE
Aggressive cows added

### PR DESCRIPTION
**Pull request by gayatri517**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

Aggressive cows program added in cp( competitive programming folder)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3215 

kindly review my PR @bhav09 
Thank you.

### CODE REVIEW

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+/* Example :
+1)Input :
+5 3
+1 2 4 8 9
+Output : 3
+
+2) Input :
+6 4
+1 3 5 7 8 10
+Output : 2  */
```
Comment (bhav09) [inline C-Plus-Plus/cp/aggressive_cows.cpp]: Add a new line here, EOF.

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+			{
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+// Time Complexity : O(nlogn)
```
Comment (bhav09) [inline C-Plus-Plus/cp/aggressive_cows.cpp]: Complexity should be at the end of the code.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

**Comment by gayatri517** [comment]
@bhav09 Changes added. Kindly check them.
Thank you.

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+1 2 4 8 9
+Output : 3
+
+2) Input :
+6 4
+1 3 5 7 8 10
+Output : 2 
+
+Time Complexity : O(nlogn)
+*/
```
Comment (bhav09) [inline C-Plus-Plus/cp/aggressive_cows.cpp:80]: Add a new line here. EOF.

**Comment by gayatri517** [comment]
@bhav09 new line added, Kindly check it.
Thank you.

**Comment by gayatri517** [comment]
Kindly have look at my PR. Thank you.
@HarshCasper

**Comment by gayatri517** [comment]
@bhav09 kindly take look at my PR. 
Thank you.

**Comment by gayatri517** [comment]
@HarshCasper @ankitaggarwal23 @siddharth25pandey 
@bhav09 please review my PR.
Thank you.

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	return false;
+}
+
+int main()
+{
+	int stalls[100];
+	int n;
+	int cows;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/aggressive_cows.cpp:36]: Display message, what are you taking input from the user

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	cin >> n >> cows;
+	//Input of the stall coordinates
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/aggressive_cows.cpp:38]: Display message, what are you taking input from the user

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		{
+			ans = mid;
+			s = mid + 1;
+		}
+		else
+		{
+			e = mid - 1;
+		}
+	}
+
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/aggressive_cows.cpp:63]: Display output message for better user-interface

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@gayatri517 Do the suggested changes

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 7]
```diff
@@ -0,0 +1,80 @@
+/*Farmer John has built a new long barn, with N (2 <= N <= 100,000) stalls.
+The stalls are located along a straight line at positions x1,...,xN (0 <= xi <= 1,000,000,000).
+His C (2 <= C <= N) cows don't like this barn layout and become aggressive towards each other once put into a stall. 
+To prevent the cows from hurting each other, FJ wants to assign the cows to the stalls, 
+such that the minimum distance between any two of them is as large as possible. 
+What is the largest minimum distance?*/
+
+#include <bits/stdc++.h>
+using namespace std;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/aggressive_cows.cpp:9]: Add a new line after using namespace std;

File: C-Plus-Plus/cp/aggressive_cows.cpp (hunk 4 above)
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/aggressive_cows.cpp:36]: Shift these variable declaration to a single line i.e `int stalls[100], n, cows;`

File: C-Plus-Plus/cp/aggressive_cows.cpp (hunk 5 above)
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/aggressive_cows.cpp:38]: Remove the redundant comments

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+	int cows;
+	cin >> n >> cows;
+	//Input of the stall coordinates
+	for (int i = 0; i < n; ++i)
+	{
+		cin >> stalls[i];
+	}
+
+	int s = 0;
+	int e = stalls[n - 1] - stalls[0];
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/aggressive_cows.cpp:45]: Use proper naming convention for declaring variables, Don't use single letters.

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+/* Example :
+1)Input :
+5 3
+1 2 4 8 9
+Output : 3
+
+2) Input :
+6 4
+1 3 5 7 8 10
+Output : 2
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/aggressive_cows.cpp:76]: Remove the extra test case one is enough

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+1 2 4 8 9
+Output : 3
+
+2) Input :
+6 4
+1 3 5 7 8 10
+Output : 2 
+
+Time Complexity : O(nlogn)
+EOF
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/aggressive_cows.cpp:79]: Remove the word EOF
Add Space Complexity

File: C-Plus-Plus/cp/aggressive_cows.cpp [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/aggressive_cows.cpp:80]: Add a new after */

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**